presto-main/src/main/java/com/facebook/presto/operator/repartition/OptimizedPartitionedOutputOperator.java

```diff
@@ -364,6 +392,7 @@ public OperatorFactory duplicate()
         private final ArrayAllocator arrayAllocator = new SimpleArrayAllocator(5000);
         private final BlockFlattener flattener = new BlockFlattener(arrayAllocator);
         private final Closer blockLeaseCloser = Closer.create();
+        private final ArrayAllocator bufferAllocator = new SimpleArrayAllocator(10000);
```

Comment:
Would it be possible to reuse existing arrayAllocator ?

- 10000 -> 10_000 for readability

Reply:
@mbasmanova The difficulty using existing arrayAllocator is counting memory becomes a bit more difficult. The existing arrayAllocator is only used by BlockFlattener, and this part of memory borrowed would be still counted in DecodedBlock#getRetainedSizeInBytes(). Thus when we add a new `bufferAllocator`, we can simply add its retained size. If we merge the two allocators, then the counting of the dictionary ids array borrowed by BlockFlattener would be double counted in arrayAllocator and decodedBlocks. How about renaming original `arrayAllocator` to  `blockDecodingAllocator` and remain using `bufferAllocator` for reusing the buffers?